Lib/http/server.py

```diff
@@ -1301,10 +1351,31 @@ def test(HandlerClass=BaseHTTPRequestHandler,
                         default='HTTP/1.0',
                         help='conform to this HTTP version '
                              '(default: %(default)s)')
+    parser.add_argument('--tls-cert', metavar='PATH',
+                        help='path to the TLS certificate')
+    parser.add_argument('--tls-key', metavar='PATH',
+                        help='path to the TLS key')
+    parser.add_argument('--tls-password-file', metavar='PATH',
+                        help='file containing the password for the TLS key')
     parser.add_argument('port', default=8000, type=int, nargs='?',
                         help='bind to this port '
                              '(default: %(default)s)')
     args = parser.parse_args()
+
+    if not args.tls_cert and args.tls_key:
+        parser.error("--tls-key requires --tls-cert to be set")
+
+    tls_key_password = None
+    if args.tls_password_file:
+        if not args.tls_cert:
+            parser.error("--tls-password-file requires --tls-cert to be set")
+
+        try:
+            with open(args.tls_password_file, "r", encoding="utf-8") as f:
+                tls_key_password = f.read().strip()
+        except (OSError, IOError) as e:
```

Comment:
```suggestion
        except OSError as e:
```

IOError is an alias to OSError